src/test/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/api/CardEventHandlerTest.java

```diff
@@ -284,4 +286,133 @@ void does_not_change_gamestate_if_its_not_the_last_card() {
+
+    assertThat(handPlayer2.getNumberOfWonTricks(), is(0));
+    assertTrue(
+        handPlayer1.getNumberOfWonTricks() == 3 && handPlayer3.getNumberOfWonTricks() == 1
+            || handPlayer1.getNumberOfWonTricks() == 1 && handPlayer3.getNumberOfWonTricks() == 3);
+  }
+
+  @Test
+  void play_last_card_with_last_two_rounds_stacked() {
+    // this test may fail in case in the battling round multiple players have the same highest cards
+    // again
+    Game game =
+        GameBuilder.builder("test", gameRepository, participationRepository, playerRepository)
+            .withParticipation(PLAYER_NAME_1)
+            .withParticipation(PLAYER_NAME_2)
+            .withParticipation(PLAYER_NAME_3)
+            .withMatch()
+            .withHandForPlayer(PLAYER_NAME_1)
+            .withCards(ACE_OF_CLUBS, QUEEN_OF_CLUBS, JACK_OF_CLUBS)
+            .finishHand()
+            .withHandForPlayer(PLAYER_NAME_2)
+            .withCards(KING_OF_CLUBS, JACK_OF_SPADES, EIGHT_OF_CLUBS)
+            .finishHand()
+            .withHandForPlayer(PLAYER_NAME_3)
+            .withCards(SEVEN_OF_CLUBS, ACE_OF_SPADES, QUEEN_OF_SPADES)
+            .finishHand()
+            .withRound()
+            .withPlayedCard(PLAYER_NAME_1, JACK_OF_CLUBS)
+            .withPlayedCard(PLAYER_NAME_2, KING_OF_CLUBS)
+            .withPlayedCard(PLAYER_NAME_3, SEVEN_OF_CLUBS)
+            .finishRound()
+            .withRound()
+            .withPlayedCard(PLAYER_NAME_1, ACE_OF_CLUBS)
+            .withPlayedCard(PLAYER_NAME_2, JACK_OF_SPADES)
+            .withPlayedCard(PLAYER_NAME_3, ACE_OF_SPADES)
+            .finishRound()
+            .withRound()
+            .withPlayedCard(PLAYER_NAME_1, QUEEN_OF_CLUBS)
+            .withPlayedCard(PLAYER_NAME_2, EIGHT_OF_CLUBS)
+            .withPlayedCard(PLAYER_NAME_3, QUEEN_OF_SPADES)
+            .finishRound()
+            .finishMatch()
+            .build();
+
+    gameRepository.saveAll(List.of(game));
+
+    List<Hand> sortedHands = getHandsSortedByParticipation(game);
+    Round battlingRound = match.getLastRound().orElseThrow();
+    Hand handPlayer1 = sortedHands.get(0);
+    Hand handPlayer2 = sortedHands.get(1);
+    Hand handPlayer3 = sortedHands.get(2);
+    simulateBattlingCardPlaying(handPlayer1, battlingRound);
+    simulateBattlingCardPlaying(handPlayer2, battlingRound);
+    simulateBattlingCardPlaying(handPlayer3, battlingRound);
+
+    assertThat(handPlayer2.getNumberOfWonTricks(), is(1));
+    assertTrue(
+        handPlayer1.getNumberOfWonTricks() == 3 && handPlayer3.getNumberOfWonTricks() == 0
+            || handPlayer1.getNumberOfWonTricks() == 0 && handPlayer3.getNumberOfWonTricks() == 3);
+  }
+
+  private List<Hand> getHandsSortedByParticipation(Game game) {
+
+    match = game.getLastMatch().get();
+    lastRound = match.getLastRound().get();
+    card1 = lastRound.getCards().iterator().next();
+    cardEventHandler.handleAfterSave(card1);
```

Comment:
? wieso wird hier der cardEventHandler aufgerufen -> das muss glaube ich geändert werden

Reply:
Aha ja also es ist ziemlich hässlich, dass ich das in einer Methode mit dem Namen getHandsSortedByParticipation mache. Ich überarbeite das.